File: Lab3/src/lab3/Tuner.java
package lab3;

import java.io.File;

import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.PrintWriter;


public class Tuner implements TvTuner, ChannelSwitcher {
    private byte loudness = 90;
    private PrintWriter fout = new PrintWriter(new FileOutputStream(
            new File("MyTvTuner.txt"),
            true));
    private boolean switchTuner = false;
    private byte channel = 0;
    private String[] channelTuner = new String[]{"QTV", "NLO TV", "Новий канал", "M1", "K1", "ICTV", "TET", "UATV"};



    public Tuner() throws FileNotFoundException {
    }
    public PrintWriter getFout() {
        return fout;
    }


    public String[] getChannelTuner() {
        return channelTuner;
    }



    @Override
    public void include() {
        if (!switchTuner) {
            switchTuner = true;
            System.out.println("Тюнер включився");
            getFout().println("Тюнер включився");
        } else {
            System.out.println("Тюнер вже був включений");
            getFout().println("Тюнер вже був включений");
        }
        getFout().flush();
    }

    @Override
    public void exclude() {
        if (switchTuner) {
            switchTuner = false;
            System.out.println("Тюнер виключили");
            getFout().println("Тюнер виключили");

        } else {
            System.out.println("Тюнер вже був виключений");
            getFout().println("Тюнер вже був виключений");
        }
        getFout().flush();
    }

    @Override
    public void change_volume_vol_add() {

        if(loudness<=100){
            System.out.println("Звук: "+(++loudness) );
            getFout().println("Звук: "+(loudness));
        }else{
            System.out.println("Вже стоїть максимальна гучність");
            getFout().println("Вже стоїть максимальна гучність");
        }
        getFout().flush();
    }

    @Override
    public void change_volume_vol_minus() {

        if(loudness>=0){
            System.out.println("Звук: "+(--loudness) );
            getFout().println("Звук: "+(loudness));
        }else{
            System.out.println("Вже стоїть мінімальна гучність");
            getFout().println("Вже стоїть мінімальна гучність");
        }
        getFout().flush();
    }




    public void switchСhannel() throws FileNotFoundException {
        if (channel == new Tuner().getChannelTuner().length) {
            channel = 0;
        }
        System.out.println("Канал: "+ new Tuner().getChannelTuner()[channel]);
        getFout().println("Канал: "+ new Tuner().getChannelTuner()[channel]);
        channel++;
        getFout().flush();
    }

    public void ChannelList() throws FileNotFoundException {
        byte num = 1;
        System.out.println("--<Список каналів>--");
        for (String el: new Tuner().getChannelTuner()){
            System.out.println("\t"+num+". " + el);
            getFout().println("\t"+num+". " + el);
            num++;
        }
        getFout().flush();

    }

    public void switchChannel() {
        if (channel == channelTuner.length) {
            channel = 0;
        }
        System.out.println("Канал: " + channelTuner[channel]);
        getFout().println("Канал: " + channelTuner[channel]);
        channel++;
        getFout().flush();
    }

    @Override
    public void switchChannel(String channelName) {
        for (byte i = 0; i < channelTuner.length; i++) {
            if (channelTuner[i].equals(channelName)) {
                channel = i;
                System.out.println("Тюнер переключився на канал: " + channelTuner[channel]);
                getFout().println("Тюнер переключився на канал: " + channelTuner[channel]);
                getFout().flush();
                return;
            }
        }
        System.out.println("Канал не знайдено: " + channelName);
        getFout().println("Канал не знайдено: " + channelName);
        getFout().flush();
    }
}

File: Lab9/Tv/Tuner.py
# Tuner.py
from TvTuner import TvTuner

class Tuner(TvTuner):
    def __init__(self):
        self.loudness = 90
        self.switch_tuner = False
        self.channel = 0
        self.channel_tuner = ["QTV", "NLO TV", "M1", "K1", "ICTV", "TET", "UATV"]
        self.fout = open("MyTvTuner.txt", "a")

    def get_fout(self):
        return self.fout

    def include(self):
        if not self.switch_tuner:
            self.switch_tuner = True
            print("Tuner is on")
            self.get_fout().write("Tuner is on\n")
        else:
            print("Tuner is almost on")
            self.get_fout().write("Tuner is almost on\n")
        self.get_fout().flush()

    def exclude(self):
        if self.switch_tuner:
            self.switch_tuner = False
            print("Tuner is off")
            self.get_fout().write("Tuner is off\n")
        else:
            print("Tuner was already on")
            self.get_fout().write("Tuner was already on\n")
        self.get_fout().flush()

    def change_volume_vol_add(self):
        if self.loudness <= 100:
            print("Volume of Tuner:", self.loudness)
            self.get_fout().write("Volume of Tuner: {}\n".format(self.loudness))
            self.loudness += 1
        else:
            print("Volume of Tuner is maximum")
            self.get_fout().write("Volume of Tuner is maximum\n")
        self.get_fout().flush()

    def change_volume_vol_minus(self):
        if self.loudness >= 0:
            print("Volume of Tuner:", self.loudness)
            self.get_fout().write("Volume of Tuner: {}\n".format(self.loudness))
            self.loudness -= 1
        else:
            print("Volume of Tuner is minimum")
            self.get_fout().write("Volume of Tuner is minimum\n")
        self.get_fout().flush()

    def switch_channel(self):
        if self.channel == len(self.channel_tuner):
            self.channel = 0
        print("Channel:", self.channel_tuner[self.channel])
        self.get_fout().write("Channel: {}\n".format(self.channel_tuner[self.channel]))
        self.channel += 1
        self.get_fout().flush()

    def channel_list(self):
        num = 1
        print("--<Channel list>--")
        for el in self.channel_tuner:
            print("\t{}. {}".format(num, el))
            self.get_fout().write("\t{}. {}\n".format(num, el))
            num += 1
        self.get_fout().flush()
